Snowflake.Data/Core/FileTransfer/StorageClient/SFRemoteStorageUtil.cs

```diff
@@ -467,7 +467,7 @@ internal static async Task DownloadOneFileAsync(SFFileMetadata fileMetadata, Can
             else
             {
                 var msg = "Unknown Error in downloading a file: " + fileMetadata.destFileName;
-                throw lastErr;
+                throw new Exception(msg);
```

Comment:
isn't that unexpected error from user perspective? or maybe we should pass cause to new exception?